committeeoversightapp/management/commands/import_data.py

```diff
@@ -115,98 +110,165 @@ def add_house_hearings(self):
+                    committees_filtered.append(subcommittee2)
+
+                try:
+                    event = Event.objects.get(
+                                            name=name,
+                                            start_date=start_date,
+                                            jurisdiction_id=self.jurisdiction_id,
+                                            classification=classification
+                                            )
+                except ObjectDoesNotExist:
+                    if name:
+                        self.smart_match(matched, name, committees_filtered, start_date, category, source, classification)
 
     def add_senate_hearings(self):
         with open('data/final/senate.csv', 'r') as csvfile:
             reader = csv.DictReader(csvfile)
-            row_count = 2
+            self.row_count = 2
 
             for row in reader:
 
-                source = row['source']
-                start_date = row['Date'].split('T', 1)[0]
+                matched = False
+
                 name = row['Hearing/Report']
+                start_date = row['Date'].split('T', 1)[0]
+                source = row['source']
                 classification = row['Type']
-                committees = [row['Committee1'], row['Committee2']]
                 category = row['Category1']
+                hearing_number_raw = row['Hearing #']
 
-                if name:
-                    # get or create hearing
-                    event, created = Event.objects.get_or_create(jurisdiction_id = self.jurisdiction_id,
-                                    name = name,
-                                    start_date = start_date,
-                                    classification = classification)
+                committees = [row['Committee1'], row['Committee2']]
+                committees_filtered = [committee for committee in committees if committee]
 
+                try:
+                    event = Event.objects.get(
+                                            name=name,
+                                            start_date=start_date,
+                                            jurisdiction_id=self.jurisdiction_id,
+                                            classification=classification
+                                            )
+                except ObjectDoesNotExist:
+                    # try to match by serial number
+                    if name and hearing_number_raw:
+                        try:
+                            hearing_number = re.search(r'\d{2,}-\d{1,}', hearing_number_raw).group(0)
+                            matched_events = Event.objects.filter(extras__hearing_number__endswith=hearing_number)
 
-                    # save committee data
-                    for committee in committees:
-                        if committee:
-                            self.new_committee(committee, event, row_count)
+                            for event in matched_events:
```

Comment:
this code seems to contemplate that we have a one-to-many match.  please add a comment about why that is okay.

Reply:
I also think that the indentation level is wrong here. Right now you are only updating the data if hearing number match.

Reply:
Adding a comment! I intended to only update the hearing at this point if there's a matching hearing number in the database—if there's not, it will try the `smart_match` function. I'm fixing the logic here to reflect that by adding `not matched` to the next if statement. Would love to learn a cleaner way to do this!